bulk_processing/bulk_processor.py

```diff
@@ -0,0 +1,173 @@
+        with db_helper.connect_to_read_replica() as self.db_connection, RabbitContext() as self.rabbit:
+
+            blobs_to_process = self.storage_client.list_blobs(self.processor.bucket_name,
+                                                              prefix=self.processor.file_prefix)
+
+            for blob_to_process in blobs_to_process:
+                self.process_bulk_file_from_bucket(blob_to_process)
+
+    def process_bulk_file_from_bucket(self, blob_to_process):
+        print(f'Processing file: {blob_to_process.name}')
+        file_to_process = self.download_file_to_process(blob_to_process)
+        success_file, error_file, error_detail_file = self.initialise_results_files(file_to_process.name)
+        successes, errors = self.process_file(file_to_process, success_file, error_file,
+                                              error_detail_file)
+
+        print(f'Uploading results from processing file: {blob_to_process.name}')
+        self.upload_results(successes, errors, success_file, error_file, error_detail_file)
+
+        print(f'Deleting remote file: {blob_to_process.name}')
+        blob_to_process.delete()
+
+        self.delete_local_files((file_to_process, success_file, error_file, error_detail_file))
+        print(f'Finished processing file: {blob_to_process.name}')
+
+    def process_file(self, file_to_process, success_file, error_file, error_detail_file):
+        try:
+            with open(file_to_process, encoding="utf-8") as open_file_to_process:
+
+                file_reader = csv.DictReader(open_file_to_process, delimiter=',')
+                header_validation_failures = self.find_header_validation_errors(file_reader.fieldnames)
+
+                if header_validation_failures:
+                    self.write_error_details_to_file([header_validation_failures], error_detail_file)
+                    return 0, 1  # success_count, error_count
+
+                return self.process_rows(file_reader, success_file, error_file, error_detail_file)
+
+        except UnicodeDecodeError as err:
+            self.write_file_encoding_error_files(err, error_file, error_detail_file, file_to_process)
+            return 0, 1  # success_count, error_count
+
+    @staticmethod
+    def write_file_encoding_error_files(err, error_file, error_detail_file, file_to_process):
+        shutil.copy(file_to_process, error_file)
+        error_detail_file.write_text(f'Invalid file encoding, requires utf-8, error: {err}')
+
+    def process_rows(self, file_reader, success_file, error_file, error_detail_file):
+        error_count = 0
+        success_count = 0
+        for line_number, row in enumerate(file_reader, 2):
+            row_errors = self.find_row_validation_errors(line_number, row)
+            if row_errors:
+                error_count += len(row_errors)
+                self.write_row_errors_to_files(row, row_errors, error_file, error_detail_file)
+            else:
+                success_count += 1
```

Comment:
why is the success count not the number of successful rows? Isn't this more like a success boolean than a count?